src/clarity-angular/tree-view/abstract-tree-selection.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2016 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+/**
+ * Author: Eudes
+ */
+
+import {TreeSelection} from "./tree-selection";
+import {TreeSelectionService} from "./providers/treeSelection.service";
+
+export abstract class AbstractTreeSelection {
+    constructor(public parent: AbstractTreeSelection,
+                public treeSelectionService: TreeSelectionService) {
+    }
+
+    public model: any;
+
+    abstract get children(): AbstractTreeSelection[];
```

Comment:
I think that by registering TreeNodes as AbstractTreeSelection providers, we can use @ContentChildren directly here and add more logic to this class so we can reuse it more easily in other components (in our case, at least Stack View and Datagrid).

Please don't fix this right now, we'll worry about it when we reuse the class.